.github/workflows/check-generated-code.yml

```diff
@@ -0,0 +1,66 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+name: Check Generated Code
+
+on:
+  # run pipeline on push event of main branch
+  push:
+    branches:
+      - main
+    paths:
+      - code_generation/**
+      - .github/workflows/*
+      - power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/**
+      - power_grid_model_c/power_grid_model_c/**
+      - src/power_grid_model/core/**
+      - pyproject.toml
+  # run pipeline on pull request
+  pull_request:
+    paths:
+      - code_generation/**
+      - .github/workflows/*
+      - power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/**
+      - power_grid_model_c/power_grid_model_c/**
+      - src/power_grid_model/core/**
+      - pyproject.toml
+  # run pipeline on merge queue
+  merge_group:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-generated-code:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9
```

Comment:
maybe bump this to `python3.12` if it remains in this action

Reply:
Following the check-code-quality action, shouldn't we use the oldest supported version in this? i.e. 3.9 (soon 3.10).

Reply:
only if we move this to the check-code-quality action. if we do it as a separate action, it doesn't really matter cause it's only generating code sections based on a template anyway